File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario1/id_151T2000Test.py
# id_151_Test.java
from scenario1.id_151 import double_the_difference

import pytest

class id_151_Test:
    """
    Test class of id_151.py.
    It contains ten unit test cases for the :func:`scenario1.id_151.double_the_difference`.
    """


    def test_double_the_difference_1(self):
        """
        Test case 1 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]) == 165
		
    def test_double_the_difference_2(self):
        """
        Test case 2 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([-1, -2, -3, -4, -5, -6, -7, -8, -9, -10]) == 0
		
    def test_double_the_difference_3(self):
        """
        Test case 3 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]) == 825
		
    def test_double_the_difference_4(self):
        """
        Test case 4 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([-1, -2, -3, -4, -5, -6, -7, -8, -9, -10, -11, -12, -13, -14, -15, -16, -17, -18, -19, -20]) == 0
		
    def test_double_the_difference_5(self):
        """
        Test case 5 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30]) == 1785
		
    def test_double_the_difference_6(self):
        """
        Test case 6 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([-1, -2, -3, -4, -5, -6, -7, -8, -9, -10, -11, -12, -13, -14, -15, -16, -17, -18, -19, -20, -21, -22, -23, -24, -25, -26, -27, -28, -29, -30]) == 0
		
    def test_double_the_difference_7(self):
        """
        Test case 7 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40]) == 3585
		
    def test_double_the_difference_8(self):
        """
        Test case 8 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([-1, -2, -3, -4, -5, -6, -7, -8, -9, -10, -11, -12, -13, -14, -15, -16, -17, -18, -19, -20, -21, -22, -23, -24, -25, -26, -27, -28, -29, -30, -31, -32, -33, -34, -35, -36, -37, -38, -39, -40]) == 0
		
    def test_double_the_difference_9(self):
        """
        Test case 9 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]) == 7185
		
    def test_double_the_difference_10(self):
        """
        Test case 10 for :func:`scenario1.id_151.double_the_difference`.
        """
        assert double_the_difference([-1, -2, -3, -4, -5, -6, -7, -8, -9, -10, -11, -12, -13, -14, -15, -16, -17, -18, -19, -20, -21, -22, -23, -24, -25, -26, -27, -28, -29, -30, -31, -32, -33, -34, -35, -36, -37, -38, -39, -40, -41, -42, -43, -44, -45, -46, -47, -48, -49, -50]) == 0
		
if __name__ == '__main__':
    pytest.main()

File: RQ2_Prompt_Elements/StarCoder_Data/HumanEvalJava_output/scenario1/id_151T2000Test.java
// DoubleTheDifferenceTest.java
package scenario1;

import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link DoubleTheDifference}.
* It contains ten unit test cases for the {@link DoubleTheDifference#doubleTheDifference(List)} method.
*/
class DoubleTheDifferenceTest {

    @Test
    void testDoubleTheDifference() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }

    @Test
    void testDoubleTheDifference2() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }

    @Test
    void testDoubleTheDifference3() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }

    @Test
    void testDoubleTheDifference4() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }

    @Test
    void testDoubleTheDifference5() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }

    @Test
    void testDoubleTheDifference6() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }

    @Test
    void testDoubleTheDifference7() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }

    @Test
    void testDoubleTheDifference8() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }

    @Test
    void testDoubleTheDifference9() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }

    @Test
    void testDoubleTheDifference10() {
        List<Object> lst = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19);
        assertEquals(100, DoubleTheDifference.doubleTheDifference(lst));
    }
}
